### PR TITLE
CIF-1546 - Pass the DOM nodes in which the components render as props

### PR DESCRIPTION
**Pull request by herzog31**
## Description

* Created new `ConfigContext` which provides mounting point selectors, page paths and GraphQL configurations.
* Removed all portals from the project and moved them to the Venia project.
* Created `Portal` component which can be used to wrap components to render them in a given selector.

See https://github.com/adobe/aem-cif-guides-venia/pull/33.

## How Has This Been Tested?

* Updated unit tests
* Manual test with Venia project

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

File: react-components/src/context/UserContext.js [hunk 1]
```diff
@@ -217,12 +218,7 @@ const UserContextProvider = props => {
     const [userCookie, setUserCookie] = useCookieValue('cif.userToken');
     const [, setCartCookie] = useCookieValue('cif.cart');
     const isSignedIn = () => !!userCookie;
-    const accountContainerQuerySelector =
-        (props.config && props.config.accountContainerQuerySelector) || '.header__accountTrigger #miniaccount';
-    const addressBookContainerQuerySelector =
-        (props.config && props.config.addressBookContainerQuerySelector) || '#addressbook';
-    const addressBookPath =
-        (props.config && props.config.addressBookPath) || '/content/venia/us/en/my-account/address-book.html';
+    const { pagePaths } = useConfigContext();
```
Comment (dplaton) [inline react-components/src/context/UserContext.js:221]: A context depending on other context is a bad idea IMO. I understand why it needs to be done here, but we need to refactor the UserContext anyway.

Comment (herzog31) [reply react-components/src/context/UserContext.js:221]: Actually it's a common pattern. But I agree. Just look at all the tests I had to adapt :D 

We would basically need to move https://github.com/adobe/aem-core-cif-components/blob/c7ff96a7a879d518d4c79b3b7795966b786b8710/react-components/src/context/UserContext.js#L316-L318 somewhere else. Maybe we can remove it from the context and have every component that uses it implement it by taking the path from the new `ConfigContext`.

Comment (dplaton) [reply react-components/src/context/UserContext.js:221]: I think that's a better approach. UserContext holds too much information as it is.

Comment (herzog31) [reply react-components/src/context/UserContext.js]: Okay, I refactored both `UserContext` and `NavigationContext` to not depend on `ConfigContext` anymore :)

**Pull request merged**